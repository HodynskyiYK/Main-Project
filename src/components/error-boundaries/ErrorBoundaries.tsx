import React, { Component, ErrorInfo } from "react"
import {IErrorBoundaryProps, IErrorBoundaryState} from './errorBoundaryTypes'

export class ErrorBoundary extends Component<IErrorBoundaryProps, IErrorBoundaryState> {
    public state: IErrorBoundaryState = {
        hasError: false
    }

    public static getDerivedStateFromError(_: Error): IErrorBoundaryState {
        // Update state so the next render will show the fallback UI.
        return { hasError: true }
    }

    public componentDidCatch(error: Error, errorInfo: ErrorInfo) {
        console.error("Uncaught error:", error, errorInfo)
    }

    public render() {
        if (this.state.hasError) {
            return (
                <div className="error-content">
                    <div className="error-card">
                        <h2 className="error-message">{this.props.errorText}</h2>
                        <button
                            className="btn error-btn"
                            onClick={() => this.setState({
                                hasError: false
                            })}
                        >Try again</button>
                    </div>
                </div>
            )
        }

        return this.props.children
    }
}