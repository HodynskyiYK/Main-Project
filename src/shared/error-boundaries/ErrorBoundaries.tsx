import React, { Component, ErrorInfo } from "react"
import {IErrorBoundaryProps, IErrorBoundaryState} from './errorBoundaryTypes'
import { Button, ButtonTypes } from '../../components/button'
import styles from './ErrorBoundary.module.scss'

export class ErrorBoundary extends Component<IErrorBoundaryProps, IErrorBoundaryState> {
    public state: IErrorBoundaryState = {
        hasError: false
    }

    public static getDerivedStateFromError(_: Error): IErrorBoundaryState {
        // Update state so the next render will show the fallback UI.
        return { hasError: true }
    }

    public componentDidCatch(error: Error, errorInfo: ErrorInfo) {
        console.error("Uncaught error:", error, errorInfo)
    }

    resetError = () => {
        this.setState({
            hasError: false
        })
    }

    public render() {
        if (this.state.hasError) {
            return (
                <div className={styles.errorContent}>
                    <div className={styles.errorCard}>
                        <h2 className={styles.errorMessage}>{this.props.errorText}</h2>
                        <Button
                            buttonClassName={ButtonTypes.ERROR_BTN}
                            buttonAction={this.resetError}
                            buttonText={'Try again'}
                        />
                    </div>
                </div>
            )
        }

        return this.props.children
    }
}