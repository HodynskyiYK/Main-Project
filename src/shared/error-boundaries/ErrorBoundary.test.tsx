import React from 'react'
import { render } from '@testing-library/react'
import { ErrorBoundary } from './ErrorBoundaries'

const Something = () => null

describe('ErrorBoundary tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<ErrorBoundary errorText={'Some error'}>
            <Something/>
        </ErrorBoundary>)
        expect(asFragment()).toMatchSnapshot()
    })
})