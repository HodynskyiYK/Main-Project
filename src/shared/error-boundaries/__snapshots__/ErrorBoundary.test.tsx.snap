// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary tests To match snapshot 1`] = `<DocumentFragment />`;
